hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -662,6 +663,61 @@ public PendingKeysDeletion getPendingDeletionKeys(final int count)
         .getPendingDeletionKeys(count, ozoneManager.getOmSnapshotManager());
   }
 
+  @Override
+  public List<Table.KeyValue<String, String>> getRenamesKeyEntries(
+      String volume, String bucket, String startKey, int count) throws IOException {
+    // Bucket prefix would be empty if volume is empty i.e. either null or "".
+    Optional<String> bucketPrefix = Optional.ofNullable(volume).map(vol -> vol.isEmpty() ? null : vol)
```

Comment:
Yeah in snapshot deletion we don't code volume and bucket will never be empty. But in case of AOS we would have to iterate through all the keys in the rename table. This function would be required in the future patches as well. We can leave this as it is for now. No harm in having the null check inside.

Reply:
We can have the checks that @hemantk-12 mentioned instead of this nested mapping, making it hard to read and simplify this.

Reply:
We should still do this, makes it more readable.